packages/block-editor/src/components/block-list/use-block-props/use-focus-first-element.js

```diff
@@ -95,6 +95,33 @@ export function useFocusFirstElement( clientId ) {
 			return;
 		}
 
+		// Check to see if Block contains focussable element before a generic caret insert.
+		if ( ! ref.current.getAttribute( 'contenteditable' ) ) {
+			const whitelistTags = [ 'a', 'button', 'td' ];
+			let focusElements;
+			let firstFocusElements = true;
+			// In the event a Block has children, focus will find the next element with tabindex which would happen to be the wrapping element of a child Block. This ensures we only focus on whitelisted tag such as link/button.
+			do {
+				// The first element may be okay to focus. Go ahead and check.
+				if ( firstFocusElements ) {
+					focusElements = focus.tabbable.findNext( ref.current );
+					firstFocusElements = false;
+				} else {
+					focusElements = focus.tabbable.findNext( focusElements );
+				}
+				// Need to make sure we're still in the current Block, if not, we could run in to trouble if focus is placed in a Block further down the page.
```

Comment:
I actually replicated this by mistake so I added a check to ensure the element was contained in the wrapping element of the Block. Otherwise, you could get dropped in a totally separate Block which would not be great.